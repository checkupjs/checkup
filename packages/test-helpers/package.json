{
  "name": "@checkup/test-helpers",
  "description": "Checkup's test helper library",
  "version": "0.0.10",
  "author": "Steve Calvert <steve.calvert@gmail.com>",
  "bugs": "https://github.com/checkupjs/checkup/issues",
  "dependencies": {
    "@checkup/core": "0.0.10",
    "fixturify-project": "^2.1.0",
    "handlebars": "^4.7.6",
    "stdout-stderr": "^0.1.13",
    "tmp": "^0.2.0"
  },
  "engines": {
    "node": ">= 12.*"
  },
  "files": [
    "/lib",
    "/static"
  ],
  "homepage": "https://github.com/checkupjs/checkup",
  "license": "MIT",
  "main": "lib/index.js",
  "publishConfig": {
    "access": "public"
  },
  "repository": "https://github.com/checkupjs/checkup",
  "scripts": {
    "build": "tsc --build",
    "test": "jest --passWithNoTests --no-cache"
  },
  "types": "lib/index.d.ts"
}
