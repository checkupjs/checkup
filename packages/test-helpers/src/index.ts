export { default as Project } from './ember-cli-fixturify-project';
export { stdout } from './stdout';
export { default as CheckupProject } from './checkup-fixturify-project';
export { default as Plugin } from './plugin';
