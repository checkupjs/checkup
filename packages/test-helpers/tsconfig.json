{
  "extends": "../../tsconfig-base.json",
  "compilerOptions": {
    "outDir": "lib",
    "rootDir": "src",
    "types": ["chai", "jest", "node"],
    "esModuleInterop": true
  },
  "references": [
    {
      "path": "../core"
    }
  ],
  "include": ["src/**/*"]
}
