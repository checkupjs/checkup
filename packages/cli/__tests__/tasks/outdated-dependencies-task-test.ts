import { CheckupProject, stdout, getTaskContext } from '@checkup/test-helpers';
import OutdatedDependenciesTask from '../../src/tasks/outdated-dependencies-task';
import OutdatedDependenciesTaskResult from '../../src/results/outdated-dependencies-task-result';

describe('outdated-dependencies-task', () => {
  let project: CheckupProject;

  beforeEach(() => {
    project = new CheckupProject('checkup-app', '0.0.0', (project) => {
      project.addDependency('react', '^15.0.0');
      project.addDependency('react-dom', '^15.0.0');
    });

    project.writeSync();
    project.gitInit();
    project.install();
  });

  afterEach(() => {
    project.dispose();
  });

  it('detects outdated dependencies and output to console', async () => {
    const result = await new OutdatedDependenciesTask(
      'meta',
      getTaskContext({
        cliFlags: { cwd: project.baseDir },
        pkg: project.pkg,
      })
    ).run();
    const taskResult = <OutdatedDependenciesTaskResult>result;

    taskResult.toConsole();

    expect(stdout()).toMatchSnapshot();
  });

  it('detects outdated dependencies as JSON', async () => {
    const result = await new OutdatedDependenciesTask(
      'meta',
      getTaskContext({
        cliFlags: { cwd: project.baseDir },
        pkg: project.pkg,
      })
    ).run();
    const taskResult = <OutdatedDependenciesTaskResult>result;

    expect(taskResult.toJson()).toMatchSnapshot();
  });
});
