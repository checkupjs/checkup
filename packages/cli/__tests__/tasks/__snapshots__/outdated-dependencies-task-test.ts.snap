// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outdated-dependencies-task detects outdated dependencies and output to console 1`] = `
"=== Outdated Dependencies ===

Major: 2
Minor: 0
Patch: 0

"
`;

exports[`outdated-dependencies-task detects outdated dependencies as JSON 1`] = `
Object {
  "meta": Object {
    "friendlyTaskName": "Outdated Dependencies",
    "taskClassification": Object {
      "category": "insights",
      "priority": "high",
    },
    "taskName": "outdated-dependencies",
  },
  "result": Object {
    "dependencies": Array [
      Object {
        "bump": "major",
        "devDependency": false,
        "easyUpgrade": false,
        "homepage": "https://reactjs.org/",
        "installed": "15.6.2",
        "isInstalled": true,
        "latest": "16.13.1",
        "mismatch": false,
        "moduleName": "react",
        "notInPackageJson": undefined,
        "notInstalled": false,
        "packageJson": "^15.0.0",
        "packageWanted": "15.6.2",
        "pkgError": undefined,
        "regError": undefined,
        "semverValid": "15.6.2",
        "unused": true,
        "usedInScripts": undefined,
      },
      Object {
        "bump": "major",
        "devDependency": false,
        "easyUpgrade": false,
        "homepage": "https://reactjs.org/",
        "installed": "15.6.2",
        "isInstalled": true,
        "latest": "16.13.1",
        "mismatch": false,
        "moduleName": "react-dom",
        "notInPackageJson": undefined,
        "notInstalled": false,
        "packageJson": "^15.0.0",
        "packageWanted": "15.6.2",
        "pkgError": undefined,
        "regError": undefined,
        "semverValid": "15.6.2",
        "unused": true,
        "usedInScripts": undefined,
      },
    ],
  },
}
`;
